### PR TITLE
Misc code and docstring cleanup

### PR DESCRIPTION
**Pull request by cdeil**
This PR contains some Mix code (PEP8) and docstring cleanup as well as trivial Python 3 fixes.

### CODE REVIEW

File: iminuit/_libiminuit.pyx [hunk 1]
```diff
@@ -78,30 +75,69 @@ cdef class Minuit:
...
-    #:map of
-    #:varname -> value
-    #:error_varname -> error
-    #:limit_varname -> limit
-    #:fix_varname -> True/False
-    #:user can just use python keyword expansion to use all the argument like
-    #:Minuit(fcn,**fitargs)
+
     cdef public object fitarg
+    """Current Minuit state in form of a dict.
```
Comment (cdeil) [inline iminuit/_libiminuit.pyx]: I'm trying to move all docstrings from the rst file to the cython file.

It's working, but one thing I noticed is that there's an extra `fitarg: object` for this member (even before my changes):
http://iminuit.readthedocs.org/en/latest/api.html#iminuit.Minuit.fitarg

Where does this come from?
How can I get rid of it?

Comment (piti118) [inline iminuit/_libiminuit.pyx]: My guess # cython: embedsignature=True, at the top of the file. You want it though.

Comment (cdeil) [inline iminuit/_libiminuit.pyx]: Also I noticed that the docstring doesn't show up in IPython help at all, only the generic, not helpful one generated by Cython:
```
In [1]: from iminuit import Minuit

In [2]: m = Minuit(lambda x: 0)
/opt/local/bin/ipython:1: InitialParamWarning: errordef is not given. Default to 1.
  #!/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/python3.4
/opt/local/bin/ipython:1: InitialParamWarning: Parameter x does not have initial value. Assume 0.
  #!/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/python3.4
/opt/local/bin/ipython:1: InitialParamWarning: Parameter x is floating but does not have initial step size. Assume 1.
  #!/opt/local/Library/Frameworks/Python.framework/Versions/3.4/bin/python3.4

In [3]: m.values
Out[3]: {'x': 0.0}

In [4]: m.values?
Type:        dict
String form: {'x': 0.0}
Length:      1
Docstring:
dict() -> new empty dictionary
dict(mapping) -> new dictionary initialized from a mapping object's
    (key, value) pairs
dict(iterable) -> new dictionary initialized as if via:
    d = {}
    for k, v in iterable:
        d[k] = v
dict(**kwargs) -> new dictionary initialized with the name=value pairs
    in the keyword argument list.  For example:  dict(one=1, two=2)
```
Any idea how to fix this?

Comment (cdeil) [inline iminuit/_libiminuit.pyx]: Also – For `values` and `errors` and `merrors` it looks OK in the Sphinx output.
Only for `fitarg` this is added, even though it's a dict like the others!?

screen shot 2015-08-01 at 09 30 45

**Comment by cdeil** [comment]
@piti118 – Does this look OK?

I'd like to continue improving the docstrings and some other things, but I'd prefer to do that in separate PRs.

**Comment by piti118** [comment]
Other than the errordef, the rest looks fine. Thank you so much :).

**Comment by piti118** [comment]
While you are at it. Lcg_Minuit.pxi should be gone.

**Comment by cdeil** [comment]
OK, I'll do the changes you mention here.

I also didn't commit the new `_libiminuit.cpp` because it's 1 MB and committing it often bloats the repo.
Sorry if this was discussed before, but I'd be +1 to remove it from the repo and require users on the dev version to have Cython installed. That's what most packages do. The releases would still contain the cpp file, i.e. normal users that install this via conda or pip don't need Cython. @piti118 What do you think? Should I remove the cpp file or add the new version here?

**Comment by cdeil** [comment]
Locally I keep getting this error:
```
docstring of iminuit.Minuit.draw_mnprofile:33: SEVERE: Unexpected section title.

Examples
--------
docstring of iminuit.Minuit.draw_mnprofile:35: WARNING: Exception occurred in plotting draw_mnprofile
 from /Users/deil/code/iminuit/doc/pyplots/draw_mnprofile.py:
Traceback (most recent call last):
  File "iminuit/_libiminuit.pyx", line 1334, in iminuit._libiminuit.Minuit._auto_frontend (iminuit/_libiminuit.cpp:19226)
NameError: name '__IPYTHON__' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/matplotlib/sphinxext/plot_directive.py", line 506, in run_code
    six.exec_(code, ns)
  File "<string>", line 7, in <module>
  File "iminuit/_libiminuit.pyx", line 296, in iminuit._libiminuit.Minuit.__init__ (iminuit/_libiminuit.cpp:3052)
  File "iminuit/_libiminuit.pyx", line 1338, in iminuit._libiminuit.Minuit._auto_frontend (iminuit/_libiminuit.cpp:19315)
ImportError: No module named 'iminuit.frontends'
docstring of iminuit.Minuit.narg:0: WARNING: Exception occurred in plotting draw_mncontour
 from /Users/deil/code/iminuit/doc/pyplots/draw_mncontour.py:
Traceback (most recent call last):
  File "iminuit/_libiminuit.pyx", line 1334, in iminuit._libiminuit.Minuit._auto_frontend (iminuit/_libiminuit.cpp:19226)
NameError: name '__IPYTHON__' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/matplotlib/sphinxext/plot_directive.py", line 506, in run_code
    six.exec_(code, ns)
  File "<string>", line 7, in <module>
  File "iminuit/_libiminuit.pyx", line 296, in iminuit._libiminuit.Minuit.__init__ (iminuit/_libiminuit.cpp:3052)
  File "iminuit/_libiminuit.pyx", line 1338, in iminuit._libiminuit.Minuit._auto_frontend (iminuit/_libiminuit.cpp:19315)
ImportError: No module named 'iminuit.frontends'
```
(a similar error with the master branch)

@piti118 Do you see that also or know how to resolve it?

**Comment by piti118** [comment]
Sure you could remove cpp file from repo. But, we should have setup.py command for distribution @ndawe taught me once long time ago. By the way, did I ever add you to iminuit pypi?

**Comment by cdeil** [comment]
OK, I'll remove the cpp file in this PR and see if / what needs to be done in setup.py.

No, I'm not added on PyPI for iminuit.

**Comment by piti118** [comment]
Hmmm... may be cython doesn't translate relative import line correctly? We could separate it into .py file then call that. That would get rid of how cython handles relative import.

ImportError: No module named 'iminuit.frontends'

**Comment by piti118** [comment]
Your username for PyPI would be?

**Comment by cdeil** [comment]
I tried changing to absolute imports and it still didn't work.
It seems to work on readthedocs, so OK for now.
I'll try to figure out a fix and make a new issue or PR.

I'm `Christoph.Deil` on PyPI.

**Comment by piti118** [comment]
You are now owner of iminuit on PyPI.

**Comment by cdeil** [comment]
@piti118 – Thank you!

**Pull request merged**